### ISSUE TITLE
Scripting - "Unable to locate script class" Error

### ISSUE THREAD

**Issue by landoncrabtree** [NationalSecurityAgency/ghidra#4519]
**Describe the bug**
When attempting to use a script (Either through Script Manager GUI or HeadlessAnalyzer postScript), it errors with "Unable to locate script class" or "Class not found", respectively.

**To Reproduce**
1. Access Scripting Manager
2. Run a script
3. Look at Console

Or if using HeadlessAnalyzer:
`analyzeHeadless.bat C:\  test -import "C:\core" -postScript "SynchronizeGDTCategoryPaths.java" -deleteProject`

I've also looked up some similar issues, where the fix was mentioning the absolute path via the -scriptPath flag. Even when passing that with `analyzeHeadless.bat C:\  test -import "C:\core" -scriptPath "C:\ghidra_10.1.5_PUBLIC\Ghidra\Features\Base\ghidra_scripts" -postScript "SynchronizeGDTCategoryPaths.java" -deleteProject` I receive the same error. 

**Expected behavior**
The script should execute.

**Screenshots**
![image](https://user-images.githubusercontent.com/34496757/183709330-37cd0ea3-54d7-453b-9c49-e71df3efc0ab.png)
![image](https://user-images.githubusercontent.com/34496757/183709914-9b07d2db-52f0-4476-974c-2a2df1141e21.png)

**Attachments**
N/A

**Environment (please complete the following information):**
- OS: Windows 10
- Java Version: 18.0.2
- Ghidra Version: 10.1.5
- Ghidra Origin: GitHub Releases

**Additional context**

**Comment by ryanmkurtz** [comment]
I was able to reproduce this in Ghidra 10.1.5 using JDK 18.  Switching to the supported JDK 11 or even JDK 17 seems to have fixed it.  I will investigate further.

**Comment by ryanmkurtz** [comment]
JDK 18 seems to work fine on a fresh 10.2-DEV build.  Probably due to the major upgrades to Apache Felix and other OSGI-related libraries.

**Comment by ryanmkurtz** [comment]
As far as I can tell this has been fixed in 10.2.  @landoncrabtree if it's possible for you to build master to verify, that would be excellent.  If you cannot, I suggest trying JDK 11 or JDK 17.  You may want to clear out your `~/.ghidra/.ghidra<version>/osgi` directory to start your script compilations off fresh after making the change.

**Comment by RenaKunisaki** [comment]
Having the same issue with 11.3, and no matter how much I mess with JAVA_HOME and JAVA_HOME_OVERRIDE, it refuses to run with any JDK other than 22.

**Comment by ryanmkurtz** [comment]
@RenaKunisaki, a few things:
* Ghidra version 11.3 does not exist.  What version are you using?  A pre-built release?  The master branch?
* Ghidra does not use `JAVA_HOME`.
* If you are using `JAVA_HOME_OVERRIDE`, you are setting it in `support/launch.properties` right (rather than using it as an environment variable)?

**Comment by RenaKunisaki** [comment]
Sorry, I meant 11.0.3. I've tried editing `launch.properties` to no avail. I have JDK 8, 11, 17, and 22 installed, but Ghidra always uses 22 regardless what I set.

I noticed it mentions that it ignores incompatible versions; I feel it should at least complain to help with diagnostics.

**Comment by TheN00bBuilder** [comment]
Spent a few hours on this issue today... for anyone else coming across this, issue #4317 claiming that you must downgrade might not be your solution - it may be an upgrade. If you built your Ghidra with a newer version of JDK (because that's what gradle found first) and give Ghidra an older one because you "assumed" it was the same one, you'll chase your tail for hours trying to find the issue :)

I also noticed that if you just delete your `~/.ghidra/` folder, and re-run the script without upgrading, you'll get an error on the exact Java API call that causes the issues between the versions. Else, you'll only get the cryptic "Unable to locate script class" issue. Not sure if it's by design, but maybe that should be listed as a bug?

**Comment by jmcarter9t** [comment]
FWIW: I am using 11.4 and had this happen today.  Since I recently upgraded, I went through all my Ghidra plugin and script projects and made sure they were linked to the same version of ghidra - 11.4.  One of the projects indicated I had a java version mismatch; I specified it should use the one Ghidra uses.  One or both of these actions seems to have fixed this today. In the past, I was missing a the ghidra_scripts directory in my home since I don't use that, so I added that -- that is a permanent fixture in my home tree, so that may not be a real solution to this problem.
